### ISSUE TITLE
On iOS the browsing history is not memorized correctly

### ISSUE THREAD

**Issue by wtlgo** [bluesky-social/social-app#2441]
**Describe the bug**

When I browse profiles via reposts, I'm unable to return back to the profile I came from when I either do the "back" gesture or press "back" button at the top of the screen. Instead it sends me directly to the root of the stack.

**To Reproduce**

1. Go to any feed
2. Open any profile (A)
3. Find a repost and go to the profile of the author (B)
4. Try to go back, it sends you back to the feed.

**Expected behavior**

I expect it to send me back to the profile I came from (A).

**Screenshots**

https://github.com/bluesky-social/social-app/assets/50925676/00eb1337-6da0-4bb1-84d2-1a002d673269

**Details**

 - Platform: iOS / iPhone X
 - Platform version: 16.7.2
 - App version: 1.62.0 (2)

**Comment by pfrazee** [comment]
Yeah the history stack on Web often does not behave the way I would expect it to

**Comment by wtlgo** [comment]
Yeah, I also noticed some weird history behavior on web version too, but I still didn't figure out what exactly happens and how to consistently replicate the problem. I was going to open the issue when I do, but it just keeps happening on me randomly.
